consensus/src/marshal/ingress/mailbox.rs

```diff
@@ -3,32 +3,47 @@ use crate::{
     types::Round,
     Block, Reporter,
 };
-use commonware_cryptography::bls12381::primitives::variant::Variant;
+use commonware_cryptography::{bls12381::primitives::variant::Variant, Digest};
 use futures::{
     channel::{mpsc, oneshot},
     SinkExt,
 };
 use tracing::error;
 
+/// An identifier for a block request.
+pub enum Identifier<D: Digest> {
+    /// The height of the block to retrieve.
+    Height(u64),
+    /// The commitment of the block to retrieve.
+    Commitment(D),
+    /// The highest finalized block. It may be the case that marshal does not have some of the
+    /// blocks below this height.
+    Tip,
+}
```

Comment:
We could re-use the [`Identifier` from `Archive`](https://github.com/commonwarexyz/monorepo/blob/7a66af61f8b7de1dbfe4a27d41830a4158385ec8/storage/src/archive/mod.rs#L15) and not have `Tip` be an option (instead the client has to first call `GetTip`)

Reply:
We could also maybe do something like:

```rs
pub enum Identifier<'a, D: Digest> {
    /// A known block, by height or digest.
    Known(archive::Identifier<'a, D>),
    /// The highest finalized block. It may be the case that marshal does not have some of the
    /// blocks below this height.
    Tip,
}

impl<'a, D: Digest> From<archive::Identifier<'a, D>> for Identifier<'a, D> {
    fn from(src: archive::Identifier<'a, D>) -> Self {
        Self::Known(src)
    }
}
```

and then change `get_block`'s signature to:

```rs
pub async fn get_block(
    &mut self,
    identifier: impl Into<Identifier<B::Commitment>>,
) -> oneshot::Receiver<Option<B>>;
```

to avoid the extra call. Having `Tip` be an option seems nice to reduce boilerplate on the consumer's end.